### PR TITLE
Implement formatter option attributePosition

### LINKED ISSUE #1706: 📎 Implement format option `singleAttributePerLine`
### Description

From the discussion https://github.com/biomejs/biome/discussions/698

It would be great if someone wanted to implement this option.

Here's the prettier reference: https://prettier.io/docs/en/options#single-attribute-per-line

I'm happy to reply to any questions you might have.

### PR DESCRIPTION
**Pull request by octoshikari**
## Summary
I've added implementation for new formatter option that will have similar behaviour  prettier option singleAttributePerLine that was added b prettier 2.6.0 version. This option last that we need in our projects to migrate from eslint to biome

Closes #1706 

## Test Plan
I've add new test cases that apply this rule

### CODE REVIEW

**Comment by tylerlaprade** [comment]
>This option last that we need in our projects to migrate from eslint to biome

To clarify, you mean Prettier here, not ESLint, right? You can separately migrate formatting/linting according to your Biome config, so migration from ESLint should not be blocked by this.

**Comment by octoshikari** [comment]
@tylerlaprade We don't use prettier, but we have this eslint-plugin-react rule `react/jsx-max-props-per-line` that similar to prettier option and make code more readable. A most easy way to do it, support Prettier option - single Attribute Per Line in biome. As result we want make full switch to biome for linting/formatting approach

File: crates/biome_js_formatter/src/context.rs [hunk 1]
```diff
@@ -169,6 +169,9 @@ pub struct JsFormatOptions {
 
     /// Information related to the current file
     source_type: JsFileSource,
+
+    /// Enforce single attribute per line in HTML, Vue and JSX. Defaults to false
```
Comment (denbezrukov) [inline crates/biome_js_formatter/src/context.rs:173]: I think it's better to remove `HTML` and `Vue` because the option is related only to JSX now.

**Review by denbezrukov — CHANGES_REQUESTED** [review]
Great PR!
Thank you for your contribution! I believe that you need to accept new snapshots for the CLI to pass the tests and remove HTML and Vue mentions.

One thought I'd like to discuss before merging: since this option is now only for JSX, and we already have a JSX-specific option for quotes (`jsx-single-quote`), I'm wondering if a better name for it might be `jsx-single-attribute-per-line`?

I understand that this differs from a Prettier option.

Any thoughts?

**Comment by denbezrukov** [comment]
I forgot to mention that we need to implement the option for the playground. We can do it in another PR if it requires many changes.

**Comment by octoshikari** [comment]
@denbezrukov Strange thing here with snapshots. I'm already updated them and you can see this in changed files, but in workflow job old version used, but i will double check this. 
For me okay  rename option for jsx only. I think this more about Prettier compability , but while biome doesn't support HTML and Vue files we can move option under jsx section, and implement in сommon formatter option after support HTML and Vue.

**Comment by denbezrukov** [comment]
Hey everyone,

Based on our discussions on discord and polls (https://github.com/biomejs/biome/discussions/1735, https://github.com/biomejs/biome/discussions/1736), we're proposing a new formatter option named `attributePosition`, with two variants:`auto` and  `multi-line`.

### Variants Explained:

`auto`: The formatter tries to fit all attributes on one line; if not possible, each attribute goes on its own line.
`multi-line`: Regardless of space, each attribute is placed on a new line for consistency.

Thanks for helping us shape this feature. Your feedback drives our project's improvement.

Cheers.

File: crates/biome_js_formatter/src/context.rs [hunk 2]
```diff
@@ -546,3 +567,24 @@ impl From<bool> for BracketSameLine {
...
     }
 }
+
+#[derive(Clone, Copy, Debug, Default, Eq, Hash, Merge, PartialEq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize, schemars::JsonSchema),
+    serde(rename_all = "camelCase")
+)]
+pub struct SingleAttributePerLine(bool);
```
Comment (denbezrukov) [inline crates/biome_js_formatter/src/context.rs:577]: Do we need it?

File: crates/biome_service/src/configuration/javascript/formatter.rs [hunk 3]
```diff
@@ -87,6 +87,15 @@ pub struct JavascriptFormatter {
     /// The type of quotes used in JavaScript code. Defaults to double.
     #[partial(bpaf(long("quote-style"), argument("double|single"), optional))]
     pub quote_style: QuoteStyle,
+
+    // it's also a top-level configurable property.
+    /// The attribute position style in JavaScript code. Defaults to auto..
```
Comment (denbezrukov) [inline crates/biome_service/src/configuration/javascript/formatter.rs:92]:
```suggestion
    /// The attribute position style in JavaScript code. Defaults to auto.
```

**Review by denbezrukov — CHANGES_REQUESTED** [review]
Great job! One thing we need to do is to update our changelog.

https://github.com/biomejs/biome/blob/main/CONTRIBUTING.md#changelog

File: crates/biome_js_formatter/tests/language.rs [hunk 4]
```diff
@@ -231,6 +247,9 @@ pub struct JsSerializableFormatOptions {
 
     /// Whether to hug the closing bracket of multiline HTML/JSX tags to the end of the last line, rather than being alone on the following line. Defaults to false.
     pub bracket_same_line: Option<bool>,
+
+    /// Enforce single attribute per line in JSX. Defaults to false
```
Comment (denbezrukov) [inline crates/biome_js_formatter/tests/language.rs:251]: It seems that we need to update the docs here.

File: crates/biome_js_formatter/src/context.rs (hunk 2 above)
Comment (octoshikari) [reply crates/biome_js_formatter/src/context.rs:577]: I forgot to remove this. 😨

**Comment by octoshikari** [comment]
@denbezrukov Changelog  and review comments has been fixed

**Review by denbezrukov — APPROVED** [review]
🎉🎉🎉

**Pull request merged**